docs/dependency-injection/dependency-injection.md

```diff
@@ -0,0 +1,206 @@
+# 1. What is Dependency Injection (DI)?
+
+Dependency injection (DI) is a design pattern that provide loosely-coupled, maintainable, and testable code. It improves modularity, maintainability, and testability, and is easily customizable and extensible.
+
+# 2. Service layer in Oryx
+
+In Oryx, dependency injection (DI) serves as the foundation for the business logic. By using DI to provide service layer, Oryx is designed to be easily customizable and extensible on every level, making it an ideal choice for projects with complex or rapidly-evolving requirements.
+
+The goal of service layer in Oryx is to  abstracts all of the system functionality,  including query backend services,  data caching and reloading, state management, and reactivity.
+
+DI is also utilized by many higher-level concepts in Oryx, ranging from adapters, normalizers,  HTTP interceptors, to authentication, product loading, cart management, and checkout process.
+
+## @spryker-oryx/di package
+
+The @spryker-oryx/di library provides dependency injection to Oryx applications, including a variety of useful utilities to streamline working with the DI. This includes:
```

Comment:
> provides dependency injection to Oryx applications

But not only to application for every other package and it can be also used outside Oryx framework.